src/parser.rs

```diff
@@ -124,3 +126,136 @@ impl Statement {
+        }
+
+        #[test]
+        fn returns_get() {
+            let statement = get_statement!("GET MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Get,
+                key: Some("MY_KEY".to_owned()),
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_get_ignoring_next_keys() {
+            let statement = get_statement!("GET KEY1 KEY2 KEY3");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Get,
+                key: Some("KEY1".to_owned()),
+                value: None
+            });
+        }
+    }
+
+    mod del {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_del_without_key() {
+            let statement = get_statement!("DEL");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_del() {
+            let statement = get_statement!("DEL MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Del,
+                key: Some("MY_KEY".to_owned()),
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_del_ignoring_next_keys() {
+            let statement = get_statement!("DEL KEY1 KEY2 KEY3");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Del,
+                key: Some("KEY1".to_owned()),
+                value: None
+            });
+        }
+    }
+
+    mod set {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_set_without_key() {
+            let statement = get_statement!("SET");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_fail_on_set_without_value() {
+            let statement = get_statement!("SET MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_set() {
+            let statement = get_statement!("SET MY_KEY MY_VALUE");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Set,
+                key: Some("MY_KEY".to_owned()),
+                value: Some("MY_VALUE".to_owned())
+            });
+        }
+
+        #[test]
+        fn returns_set_considering_all_next_values() {
+            let statement = get_statement!("SET KEY1 VALUE1 VALUE2 VALUE3");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Set,
+                key: Some("KEY1".to_owned()),
+                value: Some("VALUE1 VALUE2 VALUE3".to_owned())
+            });
+        }
+    }
+}
```

Comment:
Please add a new line, I should probably add a `cargo fmt` stage in the PR checks, right?